lib/customer.rb

```diff
@@ -0,0 +1,31 @@
+require 'csv'
+
+class Customer
+  attr_reader :id
+  attr_accessor :email, :address
+
+  def initialize(id, email, delivery_address)
+    @id = id
+    @email = email
+    @address = delivery_address
+  end
+  
+  def self.all
+    @customer = CSV.read('data/customers.csv').map { |customer| new(customer[0].to_i, customer[1], Hash(street: customer[2], city: customer[3], state: customer[4], zip: customer[5])) }
+    return @customer
```

Comment:
Since you don't use `@customer` in any other methods it shouldn't be an instance variable:

```suggestion
    return CSV.read('data/customers.csv').map { |customer| new(customer[0].to_i, customer[1], Hash(street: customer[2], city: customer[3], state: customer[4], zip: customer[5])) }
```